### DISCUSSION TITLE
image extract error

### DISCUSSION THREAD

**Discussion by 1339503169** [pymupdf/PyMuPDF#2848] [Q&A] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)

You mentioned that it is possible to extract images from a PDF using the get_text('dict') method by setting type==1, and also by using the get_images() method. In a document I am working with, the number of images extracted by these two methods is different. Is this normal? Or is there another way to determine the position of the images extracted by get_images() in the PDF?

## To Reproduce (mandatory)

import fitz
document = fitz.open('./data/problem.pdf')
page = document.load_page(0)
text_blocks = page.get_text('dict')['blocks']
img_blocks = [i for i in text_blocks if i['type'] == 1]
images = page.get_images()
print(len(img_blocks))
print(len(images))

[problem.pdf](https://github.com/pymupdf/PyMuPDF/files/13499380/problem.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
There indeed is a difference between the two ways:
* images extracted via `get_text("dict")` internally are restricted to a clip rectangle equal to the page itself: any image not completely contained in page.rect is omitted
* images reported via `page.get_image_info()` do not contain this restriction

You can adjust that by using clip=fitz.INFINITE_RECT() in the `get_text()` method.

**Comment by 1339503169** [comment]
it helps a lot

**Comment by 1339503169** [comment]
thank you
